libraries/src/Application/CMSApplication.php

```diff
@@ -904,40 +904,72 @@ public function login($credentials, $options = [])
+                        \gettype($authorisations)
+                    ),
+                    \JLog::WARNING,
+                    'api'
+                );
 
-                    // Return the error.
-                    switch ($authorisation->status) {
-                        case Authentication::STATUS_EXPIRED:
-                            $this->enqueueMessage(Text::_('JLIB_LOGIN_EXPIRED'), 'error');
+                // Set $authorisations to an empty array to avoid further issues
+                $authorisations = [];
+            }
 
+            foreach ($authorisations as $authorisation) {
+                // Ensure $authorisation is an object and has the 'status' property
+                if (\is_object($authorisation) && property_exists($authorisation, 'status')) {
+                    if ((int) $authorisation->status & $denied_states) {
+                        // Trigger onUserAuthorisationFailure Event.
+                        $dispatcher->dispatch('onUserAuthorisationFailure', new AuthorisationFailureEvent('onUserAuthorisationFailure', [
+                            'subject' => (array) $authorisation,
+                            'options' => $options,
+                        ]));
+
+                        // If silent is set, just return false.
+                        if (isset($options['silent']) && $options['silent']) {
                             return false;
+                        }
 
-                        case Authentication::STATUS_DENIED:
-                            $this->enqueueMessage(Text::_('JLIB_LOGIN_DENIED'), 'error');
+                        // Return the error.
+                        switch ($authorisation->status) {
+                            case Authentication::STATUS_EXPIRED:
+                                $this->enqueueMessage(Text::_('JLIB_LOGIN_EXPIRED'), 'error');
 
-                            return false;
+                                return false;
 
-                        default:
-                            $this->enqueueMessage(Text::_('JLIB_LOGIN_AUTHORISATION'), 'error');
+                            case Authentication::STATUS_DENIED:
+                                $this->enqueueMessage(Text::_('JLIB_LOGIN_DENIED'), 'error');
 
-                            return false;
+                                return false;
+
+                            default:
+                                $this->enqueueMessage(Text::_('JLIB_LOGIN_AUTHORISATION'), 'error');
+
+                                return false;
+                        }
                     }
+                } else {
+                    // Log a warning if the structure is unexpected
+                    \JLog::add(
+                        \sprintf(
+                            'Invalid authorisation object or missing status property in %s on line %d.',
+                            __FILE__,
+                            __LINE__
+                        ),
+                        \JLog::WARNING,
```

Comment:
```suggestion
                        Log::WARNING,
```
Use Log and not JLog.